### PR TITLE
Feature/create battle form

### PR DESCRIPTION
**Pull request by gabrielaleal**
# Description

## Trello Card

https://trello.com/c/1tAaLTv3/1-as-a-trainer-i-create-a-new-battle#

## Steps to reproduce

To install all the requirements needed:
`pip install`

To create a user:
`python backend/manage.py createsuperuser`

To run the project, you have to open two terminals. One will be running:
`python backend/manage.py runserver`

The other will be running:
`npm run start`

Now that an URL is given to you (on the python terminal), you access its `/battle/create` path and create a battle!

After that, access the `/admin` page with the super user credentials and see in Battles if the battle you created is there.

### CODE REVIEW

**Comment by gabrielaleal** [comment]
[As a Trainer, I create a new battle](https://trello.com/c/1tAaLTv3/1-as-a-trainer-i-create-a-new-battle)

**Comment by victorfsf** [comment]
Plz remove `backend/db.sqlite3` from the repository and add it to `.gitignore` to avoid pushing it everytime (if you're using sqlite3, if not, just delete the file)

File: backend/battles/models.py [hunk 1]
```diff
@@ -0,0 +1,12 @@
+from django.db import models
+
+from users.models import User  # noqa
+
+
+# Create your models here.
+class Battle(models.Model):
+    creator = models.ForeignKey(User, on_delete=models.CASCADE, related_name="battle_creator")
+    opponent = models.ForeignKey(User, on_delete=models.CASCADE, related_name="battle_opponent")
```
Comment (victorfsf) [inline backend/battles/models.py]: Did you revisit this after understanding better how `related_name` works? I'm asking that just to make sure if these are the names you want to give the relations, no need to change them if you don't want to.

File: backend/battles/views.py [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+    template_name = "create_battle.html"
+    form_class = CreateBattleForm
+    success_url = "/"
+
+    def form_valid(self, form):
+        form.instance.creator = self.request.user
+        return super().form_valid(form)
+
+    def get_initial(self):
+        return {"creator": self.request.user.id}
```
Comment (victorfsf) [inline backend/battles/views.py]: Could you name this `creator_id`? Naming it `creator` gives the idea that you're passing the whole object, not just the id.

File: backend/battles/forms.py [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+
+
+class CreateBattleForm(forms.ModelForm):
+    class Meta:
+        model = Battle
+        fields = ["opponent"]
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.fields["opponent"].queryset = User.objects.exclude(id=self.initial["creator"])
```
Comment (victorfsf) [inline backend/battles/forms.py]: Could you name this `creator_id`? Naming it `creator` gives the idea that you're passing the whole object, not just the id.

**Comment by victorfsf** [comment]
Was stopping an unauthenticated user from using the create battle form a part of this task? I can't access the trello board so I don't know

**Comment by gabrielaleal** [comment]
> Plz remove `backend/db.sqlite3` from the repository and add it to `.gitignore` to avoid pushing it everytime (if you're using sqlite3, if not, just delete the file)

Did it! Thanks.

File: backend/battles/models.py (hunk 1 above)
Comment (gabrielaleal) [reply backend/battles/models.py]: I didn't, sorry! Renamed it to "battles_as_creator" and "battles_as_opponent" (makes sense for me, but what do you think?)

**Comment by gabrielaleal** [comment]
> Was preventing an unauthenticated user from using the create battle form a part of this task? I can't access the trello board so I don't know

No, it is not relevant right now. In the future, people will only be able to create battles if they are registered and logged in.

File: backend/battles/models.py (hunk 1 above)
Comment (victorfsf) [reply backend/battles/models.py]: Sounds good! Just remember that every change on models attributes require a `makemigrations` run.

**Pull request merged**